zkevm-circuits/src/tx_circuit.rs

```diff
@@ -1690,6 +1715,138 @@ impl<F: Field> SubCircuitConfig<F> for TxCircuitConfig<F> {
+            );
+
+            // Accumulate chunk bytes RLC
+            cb.require_equal(
+                "chunk_txbytes_rlc::cur == chunk_txbytes_rlc::prev * pow_of_rand(HashLength) + HashRLC",
+                meta.query_advice(chunk_txbytes_rlc, Rotation::cur()),
+                meta.query_advice(chunk_txbytes_rlc, Rotation(-(HASH_RLC_OFFSET as i32)))
+                        * meta.query_advice(pow_of_rand, Rotation::cur())
+                        + meta.query_advice(tx_table.value, Rotation::cur()),
+            );
+
+            cb.gate(and::expr([
+                meta.query_fixed(q_enable, Rotation::cur()),
+                // Only l2 signed bytes are accumulated
+                not::expr(meta.query_advice(is_l1_msg, Rotation::cur())),
+                is_hash_rlc(meta)
+            ]))
+        });
+
+        meta.lookup_any("Correct pow_of_rand for HashLen", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                not::expr(meta.query_advice(is_l1_msg, Rotation::cur())),
+                is_hash_length(meta),
+            ]);
+
+            vec![
+                1.expr(),                                           // q_enable
+                meta.query_advice(tx_table.value, Rotation::cur()), // exponent
+                meta.query_advice(pow_of_rand, Rotation::cur()),    // pow_of_rand
+            ]
+            .into_iter()
+            .zip(pow_of_rand_table.table_exprs(meta))
+            .map(|(arg, table)| (enable.clone() * arg, table))
+            .collect()
+        });
+
+        meta.create_gate("One chunk_txbytes_len_acc, chunk_txbytes_rlc value and pow_of_rand for each tx (in fixed section)", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            // chunk_txbytes_len_acc, chunk_txbytes_rlc and pow_of_rand stay the same for the same tx
+            cb.require_equal(
+                "chunk_txbytes_len_acc' == chunk_txbytes_len_acc",
+                meta.query_advice(chunk_txbytes_len_acc, Rotation::cur()),
+                meta.query_advice(chunk_txbytes_len_acc, Rotation::prev()),
+            );
+            cb.require_equal(
+                "chunk_txbytes_rlc' == chunk_txbytes_rlc",
+                meta.query_advice(chunk_txbytes_rlc, Rotation::cur()),
+                meta.query_advice(chunk_txbytes_rlc, Rotation::prev()),
+            );
+            cb.require_equal(
+                "pow_of_rand' == pow_of_rand",
+                meta.query_advice(pow_of_rand, Rotation::cur()),
+                meta.query_advice(pow_of_rand, Rotation::prev()),
+            );
+
+            cb.gate(and::expr([
+                meta.query_fixed(q_enable, Rotation::cur()),
+                not::expr(meta.query_fixed(q_first, Rotation::cur())),
+                not::expr(is_nonce(meta)),
+                not::expr(meta.query_advice(is_calldata, Rotation::cur()))
+            ]))
+        });
+
+        meta.create_gate(
```

Comment:
I don't think you need this. We already know the "max" fixed number of txs we can assign (and also the number of fixed rows for each one of them). So effectively we can have a fixed column `q_chunk_txbytes_hash` that's set to `1` only at that specific offset where the fixed section ends?

Reply:
Removed.